### PR TITLE
Run separate signer

### PR DESCRIPTION
**Pull request by daviddrysdale**

### CODE REVIEW

File: integration/log_integration_test.sh [hunk 1]
```diff
@@ -17,21 +18,31 @@ done
...
-./trillian_log_server --private_key_password=towel --private_key_file=${TESTDATA}/log-rpc-server.privkey.pem --port ${RPC_PORT} --sequencer_sleep_between_runs="1s" --batch_size=100 &
+./trillian_log_server --private_key_password=towel --private_key_file=${TESTDATA}/log-rpc-server.privkey.pem --port ${RPC_PORT} &
 RPC_SERVER_PID=$!
 popd > /dev/null
 
 # Ensure we kill the RPC server once we're done.
 TO_KILL+=(${RPC_SERVER_PID})
 waitForServerStartup ${RPC_PORT}
 
+
```
Comment (codingllama) [inline integration/log_integration_test.sh]: nit: vertical whitespace

File: server/trillian_log_signer/main.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	}
+	go func() {
+		glog.Info("HTTP server starting")
+		http.Serve(sock, nil)
+	}()
+
+	return nil
+}
+
+func awaitSignal(cancel context.CancelFunc) {
```
Comment (codingllama) [inline server/trillian_log_signer/main.go]: Could you add a TODO to share this function between servers? I suppose it's the same everywhere.

File: server/trillian_log_signer/main.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	// Start the sequencing loop, which will run until we terminate the process. This controls
+	// both sequencing and signing.
+	// TODO(Martin2112): Should respect read only mode and the flags in tree control etc
+	ctx, cancel := context.WithCancel(context.Background())
+	go awaitSignal(cancel)
+
+	sequencerManager := server.NewSequencerManager(registry, *sequencerGuardWindowFlag)
+	sequencerTask := server.NewLogOperationManager(ctx, registry, *batchSizeFlag, *numSeqFlag, *sequencerSleepBetweenRunsFlag, util.SystemTimeSource{}, sequencerManager)
+	sequencerTask.OperationLoop()
+
```
Comment (codingllama) [inline server/trillian_log_signer/main.go]: Should you cancel() after OperationLoop returns, or are we confident the ctx is cancelled by then?

Comment (daviddrysdale) [inline server/trillian_log_signer/main.go]: `cancel` is the only way to stop the `sequencerTask`, so to get here `cancel` must have been called (by the signal handler).

File: server/trillian_log_signer/main.go (hunk 2 above)
Comment (daviddrysdale) [inline server/trillian_log_signer/main.go]: Done.

File: integration/log_integration_test.sh (hunk 1 above)
Comment (daviddrysdale) [inline integration/log_integration_test.sh]: Done.

File: server/trillian_log_server/main.go [hunk 4]
```diff
@@ -59,6 +59,7 @@ func startRPCServer(registry extension.Registry) (*grpc.Server, error) {
 	return grpcServer, nil
 }
 
+// TODO(drysdale): commonize the following 2 fns across all main()s
```
Comment (codingllama) [inline server/trillian_log_server/main.go]: Look out for conflicts with [PR 418](https://github.com/google/trillian/pull/418/commits/575bac40806fac13fffc86cd0b0cf12b6185d01e) if you go for startHTTPServer too.

Comment (daviddrysdale) [inline server/trillian_log_server/main.go]: Do you mean `startRPCServer` ? (I'm not planning to mess with that).

File: server/trillian_log_signer/main.go (hunk 2 above)
Comment (daviddrysdale) [inline server/trillian_log_signer/main.go]: Actually, I commonized -- PTAL.

File: server/trillian_log_server/main.go (hunk 4 above)
Comment (codingllama) [inline server/trillian_log_server/main.go]: Yep, `startRPCServer`. My mistake.

**Pull request merged**